### PR TITLE
Add Register/Update Endpoint for Student

### PR DESCRIPTION
**Pull request by shalearkane**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -48,6 +48,7 @@ func main() {
 	}
 
 	r.GET("/api/token/student/verify", handler.HandlerVerifyStudentIdToken)
+	r.POST("/api/student/update", handler.GinVerifyStudent, handler.HandlerUpdateStudentDetails)
```
Comment (Shubhrajyoti-Dey-FrosTiK) [inline main.go:51]: Make it a `PUT` request

File: handler/student.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	"github.com/FrosTiK-SD/auth/controller"
+	"github.com/FrosTiK-SD/auth/model"
+	studentModel "github.com/FrosTiK-SD/models/student"
+	"github.com/gin-gonic/gin"
+	"github.com/gin-gonic/gin/binding"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+func (h *Handler) HandlerUpdateStudentDetails(ctx *gin.Context) {
+	studentCollection := h.MongikClient.MongoClient.Database(constants.DB).Collection(constants.COLLECTION_STUDENT)
```
Comment (Shubhrajyoti-Dey-FrosTiK) [inline handler/student.go:14]: Why??

File: handler/student.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	updatedStudent := model.StudentPopulated{}
```
Comment (Shubhrajyoti-Dey-FrosTiK) [inline handler/student.go:15]: Make it

```go
var updatedStudent model.StudentPopulated
```

File: handler/student.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	if errBinding := ctx.ShouldBindBodyWith(&updatedStudent, binding.JSON); errBinding != nil {
+		ctx.AbortWithStatusJSON(401, gin.H{"error": errBinding.Error()})
+		return
+	}
+
+	filter := bson.M{"_id": h.Session.Student.Id, "email": h.Session.Student.InstituteEmail}
+	currentStudent := studentModel.Student{}
```
Comment (Shubhrajyoti-Dey-FrosTiK) [inline handler/student.go:22]: Same as above

File: handler/student.go [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+	}
+
+	controller.AssignUnVerifiedFields(&updatedStudent, &currentStudent)
+	controller.InvalidateVerifiedFieldsOnChange(&updatedStudent, &currentStudent)
+
+	if updateResult, errUpdate := studentCollection.ReplaceOne(ctx, filter, currentStudent); errUpdate != nil {
+		ctx.AbortWithStatusJSON(400, gin.H{"error": errUpdate.Error()})
+		return
+	} else {
+		ctx.JSON(200, gin.H{"student": updateResult})
```
Comment (Shubhrajyoti-Dey-FrosTiK) [inline handler/student.go:36]: I think all this should be shifted to the `controller`

**Review by Shubhrajyoti-Dey-FrosTiK — CHANGES_REQUESTED** [review]

File: handler/student.go (hunk 2 above)
Comment (shalearkane) [reply handler/student.go:27]: Right now ignoring this as there is no good way for Mongik to return nil results in FindOne ... rest wherever I could, I have replaced DB queries with Mongik

File: handler/student.go [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+	}
+}
+
+func (h *Handler) HandlerRegisterStudentDetails(ctx *gin.Context) {
+	idToken := ctx.GetHeader("token")
+	newStudentDetails := interfaces.StudentRegistration{}
+
+	if errBinding := ctx.ShouldBindBodyWith(&newStudentDetails, binding.JSON); errBinding != nil {
```
Comment (Shubhrajyoti-Dey-FrosTiK) [inline handler/student.go:59]: use `ctx.BindJSON`

**Review by Shubhrajyoti-Dey-FrosTiK — CHANGES_REQUESTED** [review]

**Pull request merged**